### PR TITLE
WINC-501: Upgrade to Operator SDK 1.6.1

### PR DESCRIPTION
**Pull request by sebsoto**

### CODE REVIEW

File: controllers/windowsmachine_controller.go [hunk 1]
```diff
@@ -37,6 +37,12 @@ import (
 	"github.com/openshift/windows-machine-config-operator/version"
 )
 
+//+kubebuilder:rbac:groups=machine.openshift.io,resources=machines,verbs=get;list;watch;delete
+//+kubebuilder:rbac:groups=machine.openshift.io,resources=machinesets,verbs=get;list;watch
+//+kubebuilder:rbac:groups="",resources=nodes,verbs=*
```
Comment (aravindhp) [inline controllers/windowsmachine_controller.go:42]: Do we need to add markers for the same resource in multiple places?

File: main.go [hunk 2]
```diff
@@ -23,8 +23,17 @@ import (
 	"github.com/openshift/windows-machine-config-operator/pkg/metrics"
 	"github.com/openshift/windows-machine-config-operator/pkg/nodeconfig/payload"
 	"github.com/openshift/windows-machine-config-operator/version"
+	//+kubebuilder:scaffold:imports
 )
 
+// needed to run on the hostnetwork
+//+kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,resourceNames=hostnetwork,verbs=use
```
Comment (aravindhp) [inline main.go:30]: I assume this resulted in a RBAC policy for `securitycontextconstraints`. How were we able to run on the host network without this?

File: main.go [hunk 3]
```diff
@@ -170,6 +180,7 @@ func main() {
 	if err := secretReconciler.RemoveInvalidAnnotationsFromLinuxNodes(mgr.GetConfig()); err != nil {
 		setupLog.Error(err, "error removing invalid annotations from Linux nodes")
 	}
+	//+kubebuilder:scaffold:builder
```
Comment (aravindhp) [inline main.go:183]: Why is this needed?

File: pkg/cluster/config.go [hunk 4]
```diff
@@ -16,6 +16,10 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+//+kubebuilder:rbac:groups=config.openshift.io,resources=infrastructures,verbs=get
+//+kubebuilder:rbac:groups=config.openshift.io,resources=networks,verbs=get
+//+kubebuilder:rbac:groups=operator.openshift.io,resources=networks,verbs=get
```
Comment (aravindhp) [inline pkg/cluster/config.go:21]: Can this be collapsed into one line?

File: pkg/metrics/metrics.go [hunk 5]
```diff
@@ -22,6 +22,13 @@ import (
 	"github.com/openshift/windows-machine-config-operator/pkg/nodeconfig"
 )
 
+//+kubebuilder:rbac:groups="",resources=services;services/finalizers,verbs=create;get;delete
+//+kubebuilder:rbac:groups="",resources=endpoints,verbs=create;get;delete;update;patch
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get
+//+kubebuilder:rbac:groups="",resources=nodes,verbs=list
+//+kubebuilder:rbac:groups=monitoring.coreos.com,resources=servicemonitors,verbs=get;list;delete
+//+kubebuilder:rbac:groups="",resources=events,verbs=*
+
```
Comment (aravindhp) [inline pkg/metrics/metrics.go:31]: Aren't we missing the RBAC from [windows-exporter-role.yaml](https://github.com/openshift/windows-machine-config-operator/blob/master/deploy/olm-catalog/windows-machine-config-operator/manifests/windows-exporter-role.yaml)

File: Makefile [hunk 6]
```diff
@@ -72,40 +74,38 @@ fmt: ## Run go fmt against code.
 vet: ## Run go vet against code.
 	go vet ./...
 
-ENVTEST_ASSETS_DIR=$(shell pwd)/testbin
-test: manifests generate fmt vet ## Run tests.
-	mkdir -p ${ENVTEST_ASSETS_DIR}
-	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.7.2/hack/setup-envtest.sh
-	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh; fetch_envtest_tools $(ENVTEST_ASSETS_DIR); setup_envtest_env $(ENVTEST_ASSETS_DIR); go test ./... -coverprofile cover.out
-
```
Comment (aravindhp) [inline Makefile]: Is this not needed?

File: Makefile [hunk 7]
```diff
@@ -72,40 +74,38 @@ fmt: ## Run go fmt against code.
...
+# Set the go mod vendor flags, if they're not already set
+GOFLAGS? = $(shell go env GOFLAGS)
+ifeq "$(findstring -mod=vendor,$(GOFLAGS))" "-mod=vendor"
+GO_MOD_FLAGS ?=
+else
+GO_MOD_FLAGS ?= -mod=vendor
+endif
+
+.PHONY: build
+build: generate
```
Comment (aravindhp) [inline Makefile:90]: Is there a reason not to call `fmt vet` here?

File: Makefile [hunk 8]
```diff
@@ -1,3 +1,146 @@
+# VERSION defines the project version for the bundle.
```
Comment (aravindhp) [inline Makefile]: Please add the command you ran to generate these files to the commit message. I am assuming rest of this commit need not be reviewed.

File: PROJECT [hunk 9]
```diff
@@ -0,0 +1,8 @@
+domain: windowsmachineconfig.openshift.io
```
Comment (aravindhp) [inline PROJECT:1]: Please add the command you ran to generate these files to the commit message. I am assuming rest of this commit need not be reviewed.

File: config/default/kustomization.yaml [hunk 10]
```diff
@@ -6,7 +6,7 @@ namespace: openshift-windows-machine-config-operator
 # "wordpress" becomes "alices-wordpress".
 # Note that it should also match with the prefix (text before '-') of the namespace
 # field above.
-namePrefix: windows-machine-config-operator-
+namePrefix: ""
```
Comment (aravindhp) [inline config/default/kustomization.yaml:9]: Please add a comment about why this is `""`.

File: config/default/manager_auth_proxy_patch.yaml [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+      - name: kube-rbac-proxy
+        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
```
Comment (aravindhp) [inline config/default/manager_auth_proxy_patch.yaml:13]: I assume this deployment will be removed in one of the follow up patches.

File: config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml [hunk 12]
```diff
@@ -0,0 +1,38 @@
+apiVersion: operators.coreos.com/v1alpha1
```
Comment (aravindhp) [inline config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml:1]: At which point did you run `operator-sdk generate kustomize manifests -q`? This should be present in one of the previous commit for the current commit message to make sense.

File: config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    strategy: ""
+  installModes:
+  - supported: false
+    type: OwnNamespace
+  - supported: false
+    type: SingleNamespace
+  - supported: false
+    type: MultiNamespace
+  - supported: true
+    type: AllNamespaces
```
Comment (aravindhp) [inline config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml]: I assume you will change this to:
```yaml
  installModes:
  - supported: true
    type: OwnNamespace
  - supported: false
    type: SingleNamespace
  - supported: false
    type: MultiNamespace
  - supported: false
    type: AllNamespaces
```
in a following commit.

File: bundle.Dockerfile [hunk 14]
```diff
@@ -23,21 +23,23 @@ LABEL com.redhat.openshift.versions="=v4.7"
...
+# Core bundle labels.
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=windows-machine-config-operator
+LABEL operators.operatorframework.io.bundle.channels.v1=preview,stable
+LABEL operators.operatorframework.io.bundle.channel.default.v1=stable
+LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v3
+LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.4.0+git
```
Comment (aravindhp) [inline bundle.Dockerfile:41]: Shouldn't this be `v1.5.1`?

File: bundle/manifests/manager-config_v1_configmap.yaml [hunk 15]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+data:
+  controller_manager_config.yaml: |
+    # TODO: Determine what to do with this file
```
Comment (aravindhp) [inline bundle/manifests/manager-config_v1_configmap.yaml:4]: How does this file get used? Please ping the SDK team for direction.

File: hack/common.sh [hunk 16]
```diff
@@ -40,10 +41,34 @@ OSDK_WMCO_management() {
...
-    --operator-version 0.0.0 \
+  if [[ "$COMMAND" = "run" ]]; then
+    run_wmco $OSDK_PATH
+  fi
+  if [[ "$COMMAND" = "cleanup" ]]; then
+    cleanup_wmco $OSDK_PATH
+  fi
+}
+
+run_wmco() {
```
Comment (aravindhp) [inline hack/common.sh:52]: Having `*_wmco()` and `*_WMCO()` is really confusing. I don't see an advantage in having `*_wmco()` and would prefer the code being inline.

File: Makefile [hunk 17]
```diff
@@ -1,9 +1,10 @@
-# VERSION defines the project version for the bundle. 
+# BUNDLE_VERSION defines the project version for the bundle.
```
Comment (aravindhp) [inline Makefile:1]: How about `WMCO_VERSION`?

File: docs/HACKING.md [hunk 18]
```diff
@@ -13,22 +13,30 @@ The operator image needs to be pushed to a remote repository:
...
+```
 
-# SSH key to be used for creation of cloud-private-key secret 
-export KUBE_SSH_KEY_PATH=<path to RSA type ssh key>
-
-# on AWS only:
-export AWS_SHARED_CREDENTIALS_FILE=<path to aws credentials file>
+### Deploying the operator directly
+```shell script
+make deploy IMG=<pushed_image>
```
Comment (aravindhp) [inline docs/HACKING.md:26]: Rather than `<pushed_image>`, `quay.io/<insert username>/wmco:$VERSION_TAG` might make it easier to understand which image we are talking about here.

File: docs/HACKING.md [hunk 19]
```diff
@@ -13,22 +13,30 @@ The operator image needs to be pushed to a remote repository:
...
+```
+The above command will directly deploy all the resources required for the operator to run, as well as the WMCO pod.
+To remove the installed resources:
+```shell script
+make undeploy
 ```
 
-To run the operator on a cluster, use: 
+### Deploying the operator through OLM
```
Comment (aravindhp) [inline docs/HACKING.md:34]: Are we really using OLM or the SDK here?

File: docs/HACKING.md [hunk 20]
```diff
@@ -13,22 +13,30 @@ The operator image needs to be pushed to a remote repository:
...
 ```shell script
+export VERSION=<current WMCO semver, i.e 2.0.0>
```
Comment (aravindhp) [inline docs/HACKING.md:36]: I thought this would be `BUNDLE_VERSION` or `WMCO_VERSION`.

File: docs/HACKING.md [hunk 21]
```diff
@@ -40,10 +48,15 @@ hack/olm.sh cleanup -c "<OPERATOR_IMAGE>"
 ```
 
 ### Running e2e tests on a cluster
-We need to set up all the environment variables required in [Development workflow](#development-workflow) as well as: 
+The following enviroment variables need to be set:
```
Comment (aravindhp) [inline docs/HACKING.md:51]: `must be set`

File: docs/HACKING.md [hunk 22]
```diff
@@ -40,10 +48,15 @@ hack/olm.sh cleanup -c "<OPERATOR_IMAGE>"
...
 ```shell script
+# SSH key to be used for creation of cloud-private-key secret
+export KUBE_SSH_KEY_PATH=<path to RSA type ssh key>
 export OPERATOR_IMAGE=<registry url for remote WMCO image>
 export OPENSHIFT_CI=false
+export VERSION=<current WMCO semver, i.e 2.0.0>
```
Comment (aravindhp) [inline docs/HACKING.md:57]: Same question here.

File: docs/HACKING.md [hunk 23]
```diff
@@ -123,6 +124,11 @@ operator-sdk bundle validate $BUNDLE_REPOSITORY:$BUNDLE_TAG --image-builder podm
 ```
 
 ### Creating a new operator index
+
+#### Pre-requisites
+[opm](https://github.com/operator-framework/operator-registry/) has been installed on the localhost.
```
Comment (aravindhp) [inline docs/HACKING.md:129]: Won't `build system` be more applicable here?

File: docs/HACKING.md [hunk 24]
```diff
@@ -13,22 +13,30 @@ The operator image needs to be pushed to a remote repository:
 podman push quay.io/<insert username>/wmco:$VERSION_TAG
 ```
 ## Development workflow
-To run the operator on a cluster (for testing purpose only) or to run the e2e tests for WMCO against an OpenShift 
-cluster set up, we need to setup the following environment variables.
+With the Operator image built and pushed to a remote repository, the operator can be deployed either directly,
```
Comment (aravindhp) [inline docs/HACKING.md:16]: You need to update README.md also

File: docs/HACKING.md [hunk 25]
```diff
@@ -105,8 +107,7 @@ See [the build instructions](#build) for more information on building the image.
 
 You can then run the following command in the root of this git repository:
 ```shell script
-operator-sdk bundle create $BUNDLE_REPOSITORY:$BUNDLE_TAG -d deploy/olm-catalog/windows-machine-config-operator/manifests \
---channels alpha --default-channel alpha --image-builder podman
+podman build -f bundle.Dockerfile -t <BUNDLE_REPOSITORY:$BUNDLE_TAG> .
```
Comment (aravindhp) [inline docs/HACKING.md:110]: Why can't we do `make bundle-build BUNDLE_IMG=`?

File: bundle.Dockerfile (hunk 14 above)
Comment (mansikulkarni96) [inline bundle.Dockerfile:41]: version should be updated for current operator-sdk being used.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @sebsoto 🙇🏾‍♂️  Please address my comments.

File: main.go (hunk 3 above)
Comment (jrvaldes) [reply main.go:183]: For my own education; is this just a comment or lines like this one get parsed elsewhere (annotations maybe)?

cc: @sebsoto  @aravindhp

File: bundle/manifests/manager-config_v1_configmap.yaml (hunk 15 above)
Comment (sebsoto) [reply bundle/manifests/manager-config_v1_configmap.yaml]: Changed  this and added an explanation in config/manager/kustomization.yaml

File: docs/HACKING.md (hunk 19 above)
Comment (sebsoto) [reply docs/HACKING.md:34]: OLM is being used as the operator is being deployed as a CSV. The other method deploys the operator resources directly, without the use of a bundle.

File: config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml [hunk 26]
```diff
@@ -3,36 +3,161 @@ kind: ClusterServiceVersion
...
-    capabilities: Basic Install
+    capabilities: Seamless Upgrades
+    categories: OpenShift Optional
+    certified: "false"
+    description: An operator that enables Windows container workloads on OCP
+    olm.skipRange: '>=1.0.0 <2.0.0'
+    operatorframework.io/cluster-monitoring: "true"
+    operatorframework.io/suggested-namespace: openshift-windows-machine-config-operator
+    repository: https://github.com/openshift/windows-machine-config-operator
+    support: Red Hat
```
[the remark this answers is not in the document]
Comment (mansikulkarni96) [reply config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml:14]: yes it is required for populating the date field on operatorHub with the date time stamp of the operator build in CpaaS.

File: main.go (hunk 3 above)
Comment (aravindhp) [reply main.go:183]: How about adding a comment for our benefit?

File: Makefile (hunk 6 above)
Comment (aravindhp) [reply Makefile]: Sounds good. I thought this would work out of the box.

File: Makefile [hunk 27]
```diff
@@ -89,7 +89,7 @@ endif
 
 .PHONY: build
 build: generate fmt vet
-	build/build.sh ${OUTPUT_DIR} ${GO_MOD_FLAGS}
+	build/build.sh ${OUTPUT_DIR} $(VERSION) ${GO_MOD_FLAGS}
```
Comment (aravindhp) [inline Makefile]: Adding `${VERSION}` is very confusing given it is going away in the next commit. Not sure what this achieves. This seems outside of using the new bundle.

File: hack/common.sh [hunk 28]
```diff
@@ -40,11 +41,25 @@ OSDK_WMCO_management() {
...
 
-  $OSDK_PATH $COMMAND packagemanifests \
-    --olm-namespace openshift-operator-lifecycle-manager \
-    --operator-namespace openshift-windows-machine-config-operator \
-    --operator-version 0.0.0 \
-    --timeout 5m
+  if [[ "$COMMAND" = "run" ]]; then
+    # VERSION needs to be set in order to properly deploy the operator with operator-sdk run packagemanifests.
+    # This should always line up with the version specified in bundle/windows-machine-config-operator.package.yaml.
+    # for example windows-machine-config-operator.v2.0.0 in the package file, means VERSION=2.0.0.
```
Comment (aravindhp) [inline hack/common.sh]: I though the package.yaml was deprecated. Is that not the case?

File: hack/common.sh [hunk 29]
```diff
@@ -125,18 +144,19 @@ cleanup_WMCO() {
...
 }
 
 # returns the operator version in `Version+GitHash` format
 # we are just checking the status of files that affect the building of the operator binary
 # the files here are selected based on the files that we are transferring while building the
 # operator binary in `build/Dockerfile`
+# Takes a semver as an argument. This should be the version present in the WMCO CSV.
 get_version() {
-  OPERATOR_VERSION=2.0.0
+  BASE_VERSION=$1
```
Comment (aravindhp) [inline hack/common.sh:157]: What is BASE_VERSION?

File: hack/common.sh [hunk 30]
```diff
@@ -125,18 +144,19 @@ cleanup_WMCO() {
...
-  oc delete -f deploy/namespace.yaml
+  # Remove the openshift-windows-machine-config-operator namespace
+  oc delete ns $WMCO_DEPLOY_NAMESPACE
 }
 
 # returns the operator version in `Version+GitHash` format
 # we are just checking the status of files that affect the building of the operator binary
 # the files here are selected based on the files that we are transferring while building the
 # operator binary in `build/Dockerfile`
+# Takes a semver as an argument. This should be the version present in the WMCO CSV.
```
Comment (aravindhp) [inline hack/common.sh:155]: So you are making the change but there is no change with the call sites at least in this commit. I briefly tried finding the commit where you are introducing the change and gave up 😃  And it is also confusing that `get_version()` now takes a version as input.

File: docs/HACKING.md (hunk 24 above)
Comment (aravindhp) [reply docs/HACKING.md:16]: Please update your commit message

File: docs/HACKING.md [hunk 31]
```diff
@@ -40,10 +48,15 @@ hack/olm.sh cleanup -c "<OPERATOR_IMAGE>"
...
 
 ### Running e2e tests on a cluster
-We need to set up all the environment variables required in [Development workflow](#development-workflow) as well as: 
+The following enviroment variables must be set:
 ```shell script
+# SSH key to be used for creation of cloud-private-key secret
+export KUBE_SSH_KEY_PATH=<path to RSA type ssh key>
 export OPERATOR_IMAGE=<registry url for remote WMCO image>
 export OPENSHIFT_CI=false
+export WMCO_VERSION=<current WMCO semver, i.e 2.0.0>
```
Comment (aravindhp) [inline docs/HACKING.md:57]: What happens if I set a version that is different from the yaml file?

File: docs/HACKING.md [hunk 32]
```diff
@@ -105,8 +107,7 @@ See [the build instructions](#build) for more information on building the image.
 
 You can then run the following command in the root of this git repository:
 ```shell script
-operator-sdk bundle create $BUNDLE_REPOSITORY:$BUNDLE_TAG -d deploy/olm-catalog/windows-machine-config-operator/manifests \
---channels alpha --default-channel alpha --image-builder podman
+make bundle-build BUNDLE_IMG=<BUNDLE_REPOSITORY:$BUNDLE_TAG>
```
Comment (aravindhp) [inline docs/HACKING.md:110]: Is there no way to specify the channel now?

**Review by aravindhp — CHANGES_REQUESTED** [review]
Some of the commits have nothing to do with WMCO but are prefixed with `[wmco]`. Please change appropriately. Makefile and other script changes could go with `[build]` and yaml changes can be `[deploy]`

File: main.go (hunk 3 above)
Comment (jrvaldes) [reply main.go:185]: > This is poorly documented by OperatorSDK/kubebuilder, Theres [an issue open](https://github.com/kubernetes->sigs/kubebuilder/issues/1487) to increase documentation around it.
> 

@sebsoto  thanks for adding more color here. I found [this comment](https://github.com/kubernetes-sigs/kubebuilder/issues/1487#issuecomment-823864494)   very helpful.

File: config/default/kustomization.yaml [hunk 33]
```diff
@@ -6,7 +6,9 @@ namespace: openshift-windows-machine-config-operator
 # "wordpress" becomes "alices-wordpress".
 # Note that it should also match with the prefix (text before '-') of the namespace
 # field above.
-namePrefix: windows-machine-config-operator-
+#
+# Setting this to an empty string in order to reduce resource name length
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply config/default/kustomization.yaml:10]: There was a related topic where the operator pod and binary and called `manager` by default.
But the prefix was made to `""` purely because I got feedback that the resource names were made too long.

File: bundle/tests/scorecard/config.yaml [hunk 34]
```diff
@@ -0,0 +1,49 @@
+apiVersion: scorecard.operatorframework.io/v1alpha3
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply bundle/tests/scorecard/config.yaml:1]: This does not, what ends up in the bundle is everything in `bundle/manifests` and `bundle/metadata`. Thats defined within the bundle.Dockerfile

File: bundle/windows-machine-config-operator.package.yaml [hunk 35]
```diff
@@ -1,4 +1,4 @@
 channels:
-  - currentCSV: windows-machine-config-operator.v0.0.0
+  - currentCSV: windows-machine-config-operator.v2.0.0
```
[the remark this answers is not in the document]
Comment (sebsoto) [reply bundle/windows-machine-config-operator.package.yaml:2]: This change is being made as since the version is now being declared in a central location, the Makefile, the CSV is generated with the actual version in it, and not just the v0.0.0 placeholder that it used to be.
The package.yaml has to list an actual version present in the bundle, so it needed to be changed to the current WMCO version, the only one that will be in the bundle.

File: hack/common.sh (hunk 28 above)
Comment (sebsoto) [reply hack/common.sh]: I believe it may be when trying to distribute operators using it. However, package.yaml is required for `operator-sdk run packagemanifests`, as far as I can tell this method of testing is still supported. I cannot find a depreciation warning.

File: hack/common.sh (hunk 30 above)
Comment (sebsoto) [reply hack/common.sh:155]: Split the change and the callsite into a seperate commit than this to remove confusion.

File: hack/common.sh (hunk 29 above)
Comment (sebsoto) [reply hack/common.sh:157]: Renamed to WMCO_SEMVER

File: docs/HACKING.md (hunk 31 above)
Comment (sebsoto) [reply docs/HACKING.md:57]: `operator-sdk run packagemanifests` will not work

File: docs/HACKING.md (hunk 32 above)
Comment (sebsoto) [reply docs/HACKING.md:110]: That is done within the bundle metadata

File: config/default/kustomization.yaml (hunk 33 above)
[the remark this answers is not in the document]
Comment (aravindhp) [reply config/default/kustomization.yaml:10]: Are any of the resource names changing with this transition?

[the remark this answers is not in the document]
Comment (sebsoto) [reply config/default/kustomization.yaml:10]: None should be

File: docs/HACKING.md (hunk 31 above)
Comment (sebsoto) [reply docs/HACKING.md:58]: Added documentation in the manifests creation portion

File: Makefile [hunk 36]
```diff
@@ -179,11 +179,11 @@ unit:
 # Operator-sdk is smart enough to detect vendor directory
 .PHONY: run-ci-e2e-test
 run-ci-e2e-test:
-	hack/run-ci-e2e-test.sh -t basic
+	WMCO_VERSION=$(WMCO_VERSION) hack/run-ci-e2e-test.sh -t basic
 
 .PHONY: run-ci-e2e-upgrade-test
 run-ci-e2e-upgrade-test:
-	hack/run-ci-e2e-test.sh -t upgrade
+	WMCO_VERSION=$(WMCO_VERSION) hack/run-ci-e2e-test.sh -t upgrade
```
Comment (aravindhp) [inline Makefile:186]: What is going on here and why is this needed?

File: hack/common.sh [hunk 37]
```diff
@@ -18,7 +19,7 @@ get_operator_sdk() {
 
   DOWNLOAD_DIR=/tmp/operator-sdk
   # TODO: Make this download the same version we have in go dependencies in gomod
-  wget --no-verbose -O $DOWNLOAD_DIR https://github.com/operator-framework/operator-sdk/releases/download/v0.19.4/operator-sdk-v0.19.4-x86_64-linux-gnu && chmod +x /tmp/operator-sdk || return
+  wget --no-verbose -O $DOWNLOAD_DIR https://github.com/operator-framework/operator-sdk/releases/download/v1.6.1/operator-sdk_linux_amd64 -o operator-sdk && chmod +x /tmp/operator-sdk || return
```
Comment (aravindhp) [inline hack/common.sh:22]: You should call out in the commit that this is where you are actually swapping SDK versions.

File: docs/HACKING.md [hunk 38]
```diff
@@ -70,32 +84,24 @@ Please note that you do not need to run `hack/olm.sh run` before `hack/run-ci-e2
...
+### Generating new bundle manifests
+This step should be done in the case that changes have been made to any of the RBAC requirements, or if any changes
+were made to the files in config/.
 
-If changes need to be made to the bundle spec, a new bundle can be generated with:
-```shell script
-operator-sdk generate csv --csv-version $NEW_VERSION --operator-name windows-machine-config-operator
-```
+If the operator version needs to be changed, the `WMCO_VERSION` variable at the top of the repo's Makefile should
+be set to the new version. In addition, `bundle/windows-machine-config-operator.package.yaml` should have the
```
Comment (aravindhp) [inline docs/HACKING.md:92]: Please link to the actual file

File: docs/HACKING.md [hunk 39]
```diff
@@ -40,10 +48,16 @@ hack/olm.sh cleanup -c "<OPERATOR_IMAGE>"
...
 
 ### Running e2e tests on a cluster
-We need to set up all the environment variables required in [Development workflow](#development-workflow) as well as: 
+The following enviroment variables must be set:
 ```shell script
+# SSH key to be used for creation of cloud-private-key secret
+export KUBE_SSH_KEY_PATH=<path to RSA type ssh key>
 export OPERATOR_IMAGE=<registry url for remote WMCO image>
 export OPENSHIFT_CI=false
+# This variable needs to line up with values in the Makefile and bundle/windows-machine-config-operator.package.yaml
```
Comment (aravindhp) [inline docs/HACKING.md:57]: What values? Please be specific. Sorry about being pedantic but we are going to forget all this in a few weeks and will be struggling to bump a version when it comes time to release.

**Review by aravindhp — CHANGES_REQUESTED** [review]
Almost lgtm 😃

File: Makefile (hunk 36 above)
Comment (sebsoto) [reply Makefile:186]: Variables in the Makefile arent actually exported and therefore the variable needs to be explicitly set for the command

**Review by aravindhp — APPROVED** [review]
Nice work @sebsoto. Thanks for patiently working through this migration.

/approve

**Comment by aravindhp** [comment]
@sebsoto  please attach a screenshot of a Windows node graph in lieu of a test

**Comment by sebsoto** [comment]
@aravindhp 
Node graphs:
Screenshot from 2021-04-23 10-27-43
Node details showing that it is a Windows node:
Screenshot from 2021-04-23 10-27-50

File: build/Dockerfile.ci [hunk 40]
```diff
@@ -65,7 +65,7 @@ WORKDIR /build/windows-machine-config-operator
 COPY build build
 COPY main.go .
 COPY controllers controllers
-COPY deploy deploy
+COPY bundle bundle
```
Comment (mansikulkarni96) [inline build/Dockerfile.ci:68]: don't we need to make this change in the developer Dockerfile as well?

Comment (sebsoto) [reply build/Dockerfile.ci:68]: @mansikulkarni96 you mean in `build/Dockerfile`?
The deploy directory wasnt copied as it is only needed for CI, so there is nothing to correct

**Comment by mansikulkarni96** [comment]
Thanks for working on this @sebsoto 
/lgtm
